.github/workflows/microk8s-tests.yml

```diff
@@ -22,178 +19,178 @@ jobs:
+          sudo systemctl restart docker
+          docker system info
+
+      - uses: balchua/microk8s-actions@13f73436011eb4925c22526f64fb3ecdd81289a9
+        with:
+          channel: "${{ matrix.microk8s }}"
+          # enable now to give microk8s more time to settle down.
+          addons: '["dns", "storage", "dashboard", "ingress", "metallb:10.64.140.43-10.64.140.49"]'
+          launch-configuration: "$GITHUB_WORKSPACE/.github/microk8s-launch-config-aws.yaml"
+
+      - name: Install Dependencies
+        run: |
           set -eux
-          JUJU_BUILD_NUMBER=$JUJU_BUILD_NUMBER DOCKER_USERNAME=$DOCKER_USERNAME make microk8s-operator-update
-          microk8s.ctr images list | grep juju
-          juju version --all
-
-          # clean up disk space because the runner only has 14G which is not enough.
-          go clean -cache -modcache -i -r
-          docker system prune --all --force
-        EOF
-
-    - name: Deploy Kubeflow
-      run: |
-        sg microk8s <<EOF
+          for snap in charm juju-helpers juju-wait; do
+              sudo snap install $snap --classic
+          done
+
+          sudo DEBIAN_FRONTEND=noninteractive apt update
+          sudo DEBIAN_FRONTEND=noninteractive apt install -y libssl-dev python3-setuptools
+          sudo usermod -a -G microk8s $USER
+
+      - name: Build juju and operator image
+        run: |
+          sg microk8s <<EOF
+            set -eux
+            JUJU_BUILD_NUMBER=$JUJU_BUILD_NUMBER DOCKER_USERNAME=$DOCKER_USERNAME make microk8s-operator-update
+            microk8s.ctr images list | grep juju
+            juju version --all
+
+            # clean up disk space because the runner only has 14G which is not enough.
+            go clean -cache -modcache -i -r
+            docker system prune --all --force
+          EOF
+
+      - name: Deploy Kubeflow
+        run: |
+          sg microk8s <<EOF
+            set -eux
+
+            microk8s kubectl wait --for=condition=available -nkube-system deployment/coredns deployment/hostpath-provisioner --timeout=10m
+            juju bootstrap microk8s --debug uk8s --config=caas-image-repo=$DOCKER_USERNAME --config test-mode=true --model-default test-mode=true
+
+            juju add-model kubeflow microk8s --config logging-config="<root>=DEBUG;unit=DEBUG"
+            juju deploy kubeflow-lite --trust --revision 60
+            sleep 10
+            microk8s kubectl patch role -n kubeflow istio-ingressgateway-operator -p '{"apiVersion":"rbac.authorization.k8s.io/v1","kind":"Role","metadata":{"name":"istio-ingressgateway-operator"},"rules":[{"apiGroups":["*"],"resources":["*"],"verbs":["*"]}]}'
+            sleep 6m
+            juju --debug wait -wv -m kubeflow -t 1200
+
+            microk8s kubectl get pods -l 'juju-operator' -A -o custom-columns='Name:metadata.name,Image:spec.containers[0].image'
+          EOF
+
+      - name: Test kubeflow
```

Comment:
Let's just delete this - we are better having a test like this as a ci test. For 2.9 we will never come back to fix this particular test.